File: drawbot_test/PC_Side/bitstream.pl
use Device::SerialPort;
use Time::HiRes "usleep";
#microsleeping
# Set up the serial port /dev/ttyACM0 is it on my uno, may be different for you.

my $port = Device::SerialPort->new("/dev/ttyACM0");
#sending a char(mander) hence the 8 bits
$port->databits(8);
#arduino listens at 9600 baud
$port->baudrate(9600);
$port->parity("none");
$port->stopbits(1);
# open the file
open(FILE, 'drawbot1.txt') or die "HALP, I'M DYING 'filename' [$!]\n"; 
$bitstream = <FILE>;  
close (FILE);  
sleep undef,undef,undef,0.1;
#keep a counter of which bits we have sent
my $count = 1;

#loop through the file, send one char at a time
while (count < (length $bitstream))
{
#give the arduino a chance to catch up
usleep 10000;
#grab the next char
$current=substr $bitstream,$count,1;
#send it over to arduino
$port->write($current);
$count++;
}   

File: drawbot_test/PC_Side/bitstream.py
#! /usr/bin/env python

import serial
ser = serial.Serial(port='/dev/ttyACM0', baudrate=9600, bytesize=8, parity='N', stopbits=1) #sets arduino as serial device to receive the bits

ser.open()
bitstreamId = open('drawbot1.txt', 'r') #loads bitstream as readable file
bitstream=bitstreamId.read()
count=0
for chars in bitstream:
  while(ser.inWaiting() != 0 ): #wait around for buffer to be emptied by arduino
    print ""
  ser.write(chars) #send next bit over to arduino as a character

bitstreamId.close()
print "Your object is now destroyed..."
print "I mean beautifully etched"